python/src/spark_rapids_ml/core.py

```diff
@@ -400,6 +401,57 @@ def _require_nccl_ucx(self) -> Tuple[bool, bool]:
         """
         return (True, False)
 
+    def _validate_parameters(self) -> None:
+        cls_name = self.__class__.__name__
+
+        pyspark_est: Optional[JavaParams] = None
+
+        # Is there a better way for the dynamic imports?
+        if "LogisticRegression" in cls_name:
+            from pyspark.ml.classification import LogisticRegression
```

Comment:
Think we can add a method to `_CumlClass` to hold the pyspark/cpu class type, s.t. each implementation specifies its own pyspark/cpu equivalent.

Reply:
Yeah. Good idea. Maybe just let subclasses validate themself?